webui/src/Buttons/EditButton.tsx

```diff
@@ -372,6 +373,59 @@ interface TabsSectionProps {
 	feedbacks: FeedbackInstance[]
 }
 
+interface TabStepProps{
+	selectedKey: string | false
+	k: string | number
+	i: number
+	name:string | undefined
+	inputState: {showInputEle:boolean}
+	linkClassname: string | undefined
+	socket: CompanionSocketType
+	controlId: string
+
+}
+function TabStep({selectedKey, k, i, name, inputState, linkClassname, socket, controlId }: TabStepProps){
+
+	const renameStep = useCallback(
+		(stepId: string, newName: string) => {
+			socketEmitPromise(socket, 'controls:step:rename', [controlId, stepId, newName]).catch((e) => {
+				console.error('Failed to rename step:', e)
+			})
+		},
+		[socket, controlId]
+	)
+	
+	function onChangeHandler(e: React.ChangeEvent<HTMLInputElement>){
+		setState({showInputEle: state.showInputEle});
+		if(selectedKey){
+			renameStep(selectedKey, e.target.value);
+		}
+	}
+
+	const [state, setState] = useState(inputState);
+	return (
+		<CNavItem key={k} className="nav-steps-special">
+			{state.showInputEle? (
+				<CNavLink className={linkClassname}>
+				<input
+					type="text"
+					value={name}
+					onChange={(e)=>{onChangeHandler(e)}}
```

Comment:
It would be nice to see these functions wrapped in `useCallback` like the one above is. That will help ensure react is performant by not passing it new functions each time it renders.
Saying that, it is probably overkill to do somewhere like this